ref/general/application-hardening.adoc

```diff
@@ -0,0 +1,109 @@
+For example, if the role name is `manager`, then a user who belongs to a `manager` group has access to that resource.
+This applies only when application binding information isn't specified for the application in the `ibm-application-bnd.xmi` or `server.xml` files
+
+== Web server document root
+WAR files contain application code and sensitive information.
+Only some of that information is web-servable content, so the web server document root should not be set to the WAR root.
+If the web server document root is set to the WAR root, the web server serves up all the contents of the WAR file without interpretation, meaning that code, raw JSPs, and more are served to your users.
+
+You should also never place sensitive information in the WAR root.
+The web container serves any files that are found in the root of the WAR file, which is okay if the root contains only servable content.
+However, because the web container serves files from the WAR root, you must never place any content that users shouldn't see in the WAR root.
+For example, don't put property files, class files, or other important information in there.
+If you must place such information in the WAR file, place it in the `WEB-INF` directory, as permitted by the servlet specification.
+Information in this directory is never served by the web container.
+
+== Secure servlets
+Servlets can be served by class name or with a normal URL alias.
+By default, Open Liberty does not allow you to server servlets by class name.
+Instead of defining a mapping for each servlet, a single generic URL (such as `/servlet`) serves all servlets.
+The component of the path after the base is assumed to be the class name for the servlet.
+For example, `/servlet/com.ibm.sample.MyServlet` refers to the `com.ibm.sample.MyServlet` servlet class.
+
+You can serve servlets by class name by setting the `serveServletsByClassnameEnabled` property to `true` in the `ibm-web-ext.xmi` file.
+For production purposes, don't enable this property.
+Enabling it makes it possible for anyone that knows the class name of any servlet to directly invoke it.
+
+=== Servlet aliases
+Servlets are secured by URLs.
+Each URL that is secured must be specified in the `web.xml` file.
+Servlets can have more than one alias if multiple URLs access the same servlet class.
+If a servlet has more than one alias or there are many servlets, it can be easy to forget to secure an alias.
+Because Open Liberty secures URLs and not the underlying classes, if just one servlet URL is insecure, then an intruder might be able to bypass security.
+To alleviate this problem, use wildcards to secure servlets when possible.
+If using wildcards isn't an option, carefully double check your `web.xml` file before deployment to ensure that all servlet aliases are secure.
+
+When you specify authorization constraints for servlets, you must either list no methods or list all methods for a servlet, including `GET`, `POST`, `PUT`, `HEAD`, etc.
```

Comment:
I think maybe change "including" to "which includes" here to use a more active voice

Reply:
Updated.